src/main/java/com/vinylteam/vinyl/util/impl/VinylUaParser.java

```diff
@@ -4,141 +4,241 @@
-            logger.debug("Got collection of vinyl elements {'vinylElements':{}}", vinylElements);
-
-            for (Element vinylElement : vinylElements) {
-                String release = vinylElement.getElementsByClass(classContainingVinylRelease).text();
-                String artist = vinylElement.getElementsByClass(classContainingVinylArtist).select("a").text();
-                if (artist.equals("")) {
-                    artist = "Various Artists";
-                }
-                String priceDetails = vinylElement.getElementsByClass(classContainingVinylPriceDetails).text();
-                String priceNumber = priceDetails.substring(0, priceDetails.indexOf(' '));
-                String priceCurrency = priceDetails.substring(priceDetails.indexOf(' ') + 1);
-                String vinylLink = startLink + vinylElement
-                        .getElementsByClass(classContainingVinylLinks).select("a").attr("href");
-                String[] imageLinks = vinylElement
-                        .getElementsByClass(classContainingVinylLinks).attr("style").split("'");
-                String imageLink = imageLinks[1];
-                if (!imageLink.contains("amazonaws.com")) {
-                    imageLink = "img/goods/no_image.jpg";
-                }
-                String[] linkComponents = pageLink.split("[/?]");
-                String genre = linkComponents[4];
-
-                RawOffer rawOffer = new RawOffer();
-                rawOffer.setShopId(1);
-                rawOffer.setRelease(release);
-                rawOffer.setArtist(artist);
-                rawOffer.setPrice(Double.parseDouble(priceNumber));
-                rawOffer.setCurrency(Currency.getCurrency(priceCurrency));
-                rawOffer.setGenre(genre);
-                rawOffer.setOfferLink(vinylLink);
-                rawOffer.setImageLink(imageLink);
-                rawOfferSet.add(rawOffer);
-                logger.debug("Added rawOffer to hash set{'rawOffer':{}}", rawOffer);
+            Elements offerAnchors = document.select(SELECTOR_OFFER_ANCHORS);
+            logger.debug("Got collection of offer anchors with links to offers {'offerAnchors':{}}", offerAnchors);
+
+            for (Element anchor : offerAnchors) {
+                String offerLink = START_LINK + anchor.attr("href");
+                offerLinks.add(offerLink);
+                logger.debug("Added link to hash set of offer links{'offerLink':{}}", offerLink);
             }
             logger.debug("Parsed page link {'pageLink':{}}", pageLink);
         }
+        logger.debug("Resulting hash set of offer links is {'offerLinks':{}}", offerLinks);
+        return offerLinks;
+    }
+
+    HashSet<RawOffer> readRawOffersFromAllOfferLinks(HashSet<String> offerLinks) {
+        HashSet<RawOffer> rawOfferSet = new HashSet<>();
+        for (String offerLink : offerLinks) {
+            RawOffer rawOffer = getRawOfferFromOfferLink(offerLink);
+            if (isValid(rawOffer)) {
```

Comment:
Method isValid() will return true if only one of conditions is true. But you should check each of conditions for valid this raw offer